examples/knx-linux-coupler/main.cpp

```diff
@@ -0,0 +1,112 @@
+#include "knx_facade.h"
+
+#include "knx/bau091A.h"
+#include "knx/bau2920.h"
+
+#include "knx/bits.h"
+#include <time.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <signal.h>
+#include <sched.h>
+#include <sys/mman.h>
+
+#include "fdsk.h"
+
+volatile sig_atomic_t loopActive = 1;
+void signalHandler(int sig)
+{
+  (void)sig;
+
+  // can be called asynchronously
+  loopActive = 0;
+}
+
+bool sendHidReport(uint8_t* data, uint16_t length)
+{
+    return false;
+}
+bool isSendHidReportPossible()
+{
+    return false;
+}
+
+#if MASK_VERSION == 0x091A
+KnxFacade<LinuxPlatform, Bau091A> knx; // IP/TP1 coupler
+#elif MASK_VERSION == 0x2920
+KnxFacade<LinuxPlatform, Bau2920> knx; // TP1/RF coupler
+#else
+#error Mask version not supported yet!
+#endif
+
+void appLoop()
+{
+    if (!knx.configured())
+        return;
+}
+
+void setup()
+{
+    srand((unsigned int)time(NULL));
+    knx.readMemory();
+
+    if (knx.induvidualAddress() == 0)
+        knx.progMode(true);
+
+    if (knx.configured())
+    {
+        printf("configured %d\n", knx.paramByte(5));
+    }
+    else
+        println("not configured");
+    knx.start();
+}
+
+int main(int argc, char **argv)
+{
+    printf("main() start.\n");
+
+    uint8_t serialNumber[] = { 0x00, 0xFA, 0x01, 0x02, 0x03, 0x04};
+    uint8_t key[] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F};
+
+    FdskCalculator calc;
+    char fdskString[42]; // 6 * 6 chars + 5 dashes + nullbyte = 42
+    calc.snprintFdsk(fdskString, sizeof(fdskString), serialNumber, key);
+    printf("FDSK: %s\n", fdskString);
+
+    // Prevent swapping of this process
```

Comment:
Just out of interest: why is it necessary to prevent swapping?

Reply:
This was introduced when I added KNX-RF. As we only have the single main loop and RF timing is critical, I have added this to make sure that any process memory is swapped to disk.

Reply:
I have no proof that this really has any effect. As KNX-RF with a CC1101 directly connected via SPI is really ugly. The RPi causes so much interference the RX sensitivity is really bad. Ok, with a proper PCB layout and some filtering this could be handled, but I am not that much interested in using the CC1101 directly with the RPi. It served the purpose of being a dev platform to test things quickly related to the CC1101.